snowpack/assets/require-or-import.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+const { pathToFileURL } = require('url');
+const NATIVE_REQUIRE = eval('require');
+const NATIVE_IMPORT = (filepath) => import(filepath);
+
+/**
+ * A utility function to use Node's native `require` or dynamic `import` to load CJS or ESM files
+ * @param {string} filepath 
+ */
+module.exports = async function requireOrImport(filepath) {
+    return new Promise((resolve, reject) => {
+        try {
+            let mdl = NATIVE_REQUIRE(filepath);
```

Comment:
Is there a reason to use require() at all? You should be able to import both ESM and CJS from `import()`, So I would think trying require is unnecessary. Let me know if I'm missing something though.

Reply:
Huh, TIL! The only argument for keeping it would be to warn users who may not be on LTS yet. Even if we drop support for `node@10` there might still be `node@12` users stuck on something before `node@12.17.x`?

Reply:
I'm not following how the warning can even work. If the Node version doesn't support ESM doesn't that mean it will throw as a ParseError and never get here?

Reply:
Did a little test and I get a runtime error so it might be ok:

```js
if(true) {
	async function go() {
		const r = await import('./other.mjs')
	}

	go();
}
```

Reply:
Running `import` would probably throw in a different way, but by running `require` first it attempts to parse the file as CJS and throws a `SyntaxError` when it encounters `export` or `import` in the ESM file.